### PR TITLE
#319 Eliminated proxy reference from Unicomplex and docs

### PR DESCRIPTION
**Pull request by akara**
Thanks for your pull request.  Please review the following guidelines.

- [X] Title includes issue id.
- [X] Description of the change added.
- [X] Commits are squashed.
- [X] Tests added.
- [X] Documentation added/updated.
- [X] Also please review [CONTRIBUTING.md](https://github.com/paypal/squbs/blob/master/CONTRIBUTING.md).

... except for squbs-pipeline.

### CODE REVIEW

File: docs/bootstrap.md [hunk 1]
```diff
@@ -192,9 +192,7 @@ The web-context is a string that uniquely identifies the web context of a reques
 
 Optionally, the listeners parameter declares a list of listeners to bind this service. Listener binding is discussed in the following section, below.
 
-A proxy is a pipelined request pre-processor before the request gets processed by the request handler. The proxy name can be specified by a `proxy-name` parameter. Unless `""` is specified, a default proxy will be used. Please refer to [Request/Response Pipeline Proxy](pipeline.md) for more information.
-
-The proxy is only valid when using Spray. Please see the [pipeline configuration](streamingpipeline.md) for Akka Http experimental mode uses.
+The pipeline is a set of request pre-processors before the request gets processed by the request handler. The pipeline name can be specified by a `pipeline` parameter. Unless `""` is specified, a default pipeline will be used. Please refer to [Streaming Request/Response Pipeline](streamingpipeline.md) for more information.
```
Comment (anilgursel) [inline docs/bootstrap.md]: > Unless `""` is specified, a default pipeline will be used.

This is no longer true.  `""` is an invalid pipeline name.  To turn off default, need to explicitly do it with `defaultPipelineOn = false` setting.  Between, I had created https://github.com/paypal/squbs/issues/315 to change the setting to `defaultPipeline = on`

File: squbs-unicomplex/src/test/scala/org/squbs/unicomplex/InvalidPipelineFlowSpec.scala [hunk 2]
```diff
@@ -25,16 +25,16 @@ import org.scalatest.{FlatSpecLike, Matchers}
 import scala.concurrent.Await
 import scala.util.Failure
 
-object InvalidProxyFlowSpec {
+object InvalidPipelineFlowSpec {
 
   val dummyJarsDir = getClass.getClassLoader.getResource("classpaths").getPath
 
-  val classPath = dummyJarsDir + "/InvalidProxyFlowSvc/META-INF/squbs-meta.conf"
+  val classPath = dummyJarsDir + "/InvalidPipelineFlowSvc/META-INF/squbs-meta.conf"
```
Comment (anilgursel) [inline squbs-unicomplex/src/test/scala/org/squbs/unicomplex/InvalidPipelineFlowSpec.scala]: Can we put the file under `/InvalidPipelineFlowSpec/META-INF/squbs-meta.conf"` instead ?

Comment (akara) [inline squbs-unicomplex/src/test/scala/org/squbs/unicomplex/InvalidPipelineFlowSpec.scala]: Sure can. I think we don't have a clear convention here. In most cases it is `Svc` and in some cases it is `Spec`. Please take a look at `squbs-unicomplex/src/main/resources/classpaths` and lets discuss before I make this change.

Comment (anilgursel) [inline squbs-unicomplex/src/test/scala/org/squbs/unicomplex/InvalidPipelineFlowSpec.scala]: I try to match it with the test name just to make it easier to find the files.

Comment (akara) [inline squbs-unicomplex/src/test/scala/org/squbs/unicomplex/InvalidPipelineFlowSpec.scala]: There are a couple of places where the same entry is used by multiple tests. Shall we say we move the rest to a `Spec` style directory name? If so we may want to do so across the board.

Comment (anilgursel) [inline squbs-unicomplex/src/test/scala/org/squbs/unicomplex/InvalidPipelineFlowSpec.scala]: If it is used in multiple places, then ok.

**Pull request merged**